pymatgen/core/structure.py

```diff
@@ -3178,21 +3178,30 @@ def rotate_sites(self, indices=None, theta=0, axis=None, anchor=None,
                 properties=site.properties)
             self._sites[i] = new_site
 
-    def perturb(self, distance):
+    def perturb(self, distance, min_random_distance=None):
         """
         Performs a random perturbation of the sites in a structure to break
         symmetries.
 
         Args:
             distance (float): Distance in angstroms by which to perturb each
                 site.
+            min_random_distance (None, int, or float): If None, perturb
+                each site by 'distance'. If int or float, perturb each site a
+                distance drawn from the uniform distribution between
+                'min_random_distance' and 'distance'.
+
         """
 
         def get_rand_vec():
             # deals with zero vectors.
             vector = np.random.randn(3)
             vnorm = np.linalg.norm(vector)
-            return vector / vnorm * distance if vnorm != 0 else get_rand_vec()
+            if isinstance(min_random_distance, (float, int)):
```

Comment:
In fact this whole thing could be rewritten much more nicely as:

````python
 if isinstance(min_random_distance, (float, int)):
    distance = np.random.uniform(min_random_distance, distance)

return vector / vnorm * distance if vnorm != 0 else get_rand_vec()
```

Reply:
So for some reason when I tried that code, it treats `distance` inside `np.random.uniform(min_random_distance, distance)` as an unresolved variable local to get_rand_vec(). Would this be an ok solution:

```python
dist = distance
if isinstance(min_random_distance, (float, int)):
     dist = np.random.uniform(min_random_distance, dist)
return vector / vnorm * dist if vnorm != 0 else get_rand_vec()
```